modules/standard/OS.chpl

```diff
@@ -857,4 +964,20 @@ module OS {
     extern proc write(fildes:c_int, buf:c_void_ptr, size:c_size_t):c_ssize_t;
 
   } // end POSIX
+
+  /*Check whether or not the environment variable ``name`` is defined.
+    If ``name`` is defined then return 1 and update ``string_out``
+    to store the value of the environment variable
+    otherwise the function returns 0.
+
+    :arg name: name of the environment variable
+    :type name: `c_string`
+
+    :arg string_out: store the value of ``name`` environment variable if defined
+    :type string_out: `c_string`
+
+    :returns: 1 if ``name`` is defined and 0 if not
+    :rtype: `c_int`
+   */
+  extern proc sys_getenv(name:c_string, ref string_out:c_string):c_int;
```

Comment:
This is pointing to a wrapper we defined in our runtime for `getenv`, rather than being an extern declaration of the OS call directly.  Was it used somewhere in this form?  I'm not seeing a test that had to get updated.  My instinct is that it should be a direct wrapper of `getenv` and match that definition as much as possible if it's going to live in the OS module.

Also, I think `getenv` is a POSIX function so it should be in the POSIX module?  `setenv` is in the POSIX docs and links to a version of `getenv` there

Reply:
It's called in this form in a couple places in Path and Subprocess. 

We already have the direct extern declaration of the `getenv` system call in OS.POSIX, so maybe it would be best to completely remove the runtime-referencing declaration and convert the Path and Subprocess calls to use `OS.POSIX.getenv()` directly?

Reply:
Unless it's much nicer to have the "return 0 if it's not found" interface available?

Reply:
I think it'd be good to remove the `sys_getenv` declaration in OS, but I think it would be reasonable for Path and Subprocess to still use it, they'd just want to add their own extern declaration of it where they want to use it